packages/preact/src/index.ts

```diff
@@ -465,6 +472,29 @@ export function useSignalEffect(
 	}, []);
 }
 
+function JSXBind({ cb }: { cb: () => unknown }) {
+	return h(SignalValue, {
+		data: useComputed(cb),
+	});
+}
+
+const jsxBindPrototype = Object.getOwnPropertyDescriptors({
+	constructor: undefined,
+	type: JSXBind,
+	get props() {
+		return this;
+	},
+});
+
+/**
+ * Bind the given callback to a JSX attribute or JSX child. This allows for "inline computed"
+ * signals that derive their value from other signals. Like with `useComputed`, any non-signal
+ * values used in the callback are captured at the time of binding and won't change after that.
+ */
+export function jsxBind<T>(cb: () => T): T {
+	return Object.defineProperties({ cb }, jsxBindPrototype) as any;
```

Comment:
Why don't we re-use the existing https://github.com/preactjs/signals/blob/eab99b182985e10a237c53c8a900dbd78e5709d3/packages/preact/src/index.ts#L160-L173 which has the proven way to do this. Currently you _can_ return JSX elements which will exhibit the behaviour we fix in those descriptors

Reply:
Done 👍